### PR TITLE
Miscellaneous Tooling Updates

### PR DESCRIPTION
**Pull request by lalli-flores**
Miscellaneous tooling updates to standardize on previous team agreements.
Includes:
- Removing `prettier` as dev dependency
- Updating Node.js version to active LTS
- Standardizing on NPM ignore files
- Added dependency guide that explains what and how each dependency in the repository is used

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -1,2 +0,0 @@
-*
-!dist/*
```
Comment (adamweeks) [inline .npmignore:2]: img

File: dependencies.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
+# Dependencies
```
Comment (adamweeks) [inline dependencies.md:1]: Nice! Let's make sure to update as deps are added though

File: .npmignore (hunk 1 above)
Comment (lalli-flores) [reply .npmignore:2]: We talked about this before? We had agreed to use `files` in `package.json` instead like in the [widgets repo](https://github.com/webex/widgets/blob/master/package.json#L26)

File: dependencies.md (hunk 2 above)
Comment (lalli-flores) [reply dependencies.md:1]: Yes! That was the intent.
We need to be more conscious of the things we pull in, as we have to maintain them (update + security fixes). I felt that by explaining what dependencies are and what they do for us can help our future selves remember why we added them and be able to much more readily evaluate if they're still necessary in the future.

File: .npmignore (hunk 1 above)
Comment (adamweeks) [reply .npmignore:2]: I honestly don't remember which way we ended up but as long as it is consistent, I'm fine with.

**Pull request merged**